File: GO/functions/is_prime.go
package main
import "fmt"

func is_prime (num int) int {
	b := num - 1
	for (b > 1) {
		if (num % b == 0) {
			return 0
		}
	b -= 1
	}
return 1
}

func main () {
	var x int
	var results int
	fmt.Println("User, enter a number: ")
	fmt.Scan(&x)
	results = is_prime(x)
	if (results == 1) {
		fmt.Printf("%d IS a prime\n", x)
	} else {
	fmt.Printf("%d is NOT a prime\n", x)
	}
}

File: PYTHON/Pure Functions/is prime.py
def is_prime (num):
    num = int(num)
    b = int(num - 1)
    while (b > 1):
        if (num % b == 0):
            return 0
        b = int(b - 1)
    return 1

x = input("User, enter a number:\n")
x = int(x)

results = is_prime(x)
if (results == 1):
    print("Is prime")
else:
    print("Is NOT prime")
input("\nDONE!\n")
